src/main/java/hudson/plugins/brakeman/scanners/BrakemanJSONScanner.java

```diff
@@ -0,0 +1,84 @@
+package hudson.plugins.brakeman.scanners;
+
+import hudson.plugins.analysis.core.ParserResult;
+import hudson.plugins.analysis.util.model.Priority;
+import hudson.plugins.brakeman.Warning;
+
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+/**
+ * A Java class that represents a JSON Scanner of the Brakeman Output file
+ * This class includes a specific strategy for implementing the parsing process
+ * of the output file.
+ *
+ * @author Thomas Baird
+ */
+public class BrakemanJSONScanner implements AbstractBrakemanScanner {
+    /**
+     * Creates a new instance of <code>BrakemanJSONScanner</code>
+     */
+    public BrakemanJSONScanner() {}
+
+    public void scan(String content, ParserResult project) {
+        this.scanWarnings(content, project);
+        this.scanIgnoredWarnings(content, project);
+    }
+
+    private void scanWarnings(String brakemanOutput, ParserResult project) {
+        try {
+            JSONObject brakemanFile = new JSONObject(brakemanOutput);
+            scanJSONResultSet(brakemanFile, project, "warnings");
+        } catch(JSONException e) {
+            e.printStackTrace();
+        }
+    }
+
+    private void scanIgnoredWarnings(String brakemanOutput, ParserResult project) {
+        try {
+            JSONObject brakemanFile = new JSONObject(brakemanOutput);
+            scanJSONResultSet(brakemanFile, project, "ignored_warnings");
+        } catch (JSONException e) {
+            e.printStackTrace();
+        }
+    }
+
+    private void scanJSONResultSet(JSONObject brakemanResultFile, ParserResult project, String filterType) {
+        try {
+            JSONArray rows = brakemanResultFile.getJSONArray(filterType);
+            for (int i = 0; i < rows.length(); i++) {
+                JSONObject row = rows.getJSONObject(i);
+                String fileName = row.getString("file");
+                int line = row.getInt("line");
+                String type = row.getString("warning_type");
+                String category;
+                if (filterType == "ignored_warnings") {
+                    category = "Ignored";
+                } else {
+                    category = "General";
+                }
+                String message = row.getString("message") + ": " + row.getString("code");
```

Comment:
Dude ... i had to look this up.  Do not use `StringBuffer` that object is synchronized and you don't need thread safety in a local method.  Use a `StringBuilder` instead.

I have fogots mucho jabba.